pandas/core/arrays/categorical.py

```diff
@@ -103,7 +103,10 @@ def func(self, other):
             mask = (self._codes == -1) | (other_codes == -1)
             if mask.any():
                 # In other series, the leads to False, so do that here too
-                ret[mask] = False
+                if opname == "__ne__":
+                    ret[mask & (self._codes == other_codes)] = True
```

Comment:
This redundancy was intentional and an attempt to "reuse" the `mask` calculation above since I think it may short-circuit when `mask` is `False`, but could just make it `(self._codes == -1) & (other_codes == -1)` as well. I also think `__ne__` is the only situation that'd have to be special-cased here out of the comparison operators. Also worth pointing out that this is assuming we're dealing with `NaN` rather than `NA`.

Reply:
> `(self._codes == -1) & (other_codes == -1)`

I think this would be more clear.  Also there might be a cached `_isnan` attribute for this

Reply:
@dsaxton im working on cleaning up these comparisons, am confused by this line.  can we make this use the more common pattern

```
fill_value = True if op is operator.ne else False
[...]

ret = op(self._codes, other_codes)
mask = (self._codes == -1) | (other_codes == -1)
ret[mask] = fill_value
```

Reply:
@jbrockmendel That is much simpler and seems correct to me